### DISCUSSION TITLE
How to deploy the Dockerfile w/ specific port

### DISCUSSION THREAD

**Discussion by invisprints** [zeabur/.github#4] [Q&A] [answered]
Recently I want to deploy the https://github.com/Chanzhaoyu/chatgpt-web but meet a problem. Everything works well when I deploy the service, but I can’t visit the web reporting “Service Crashed”. It seems the service is not listening on the port of environment variable $PORT. But how to set the port? I tried to set the port to 3002 but it doesn’t work.

**Comment by yuaanlin** [comment]
Hi, the port is not 3002. You should use $PORT (environment variable) to get the right port.

For example, assume your project is a Node.js service:

This is wrong: app.listen(3002)

This is correct: app.listen(process.env.PORT)

**Comment by invisprints** [comment]
Does it mean I need to manual modify some files in the repo? I am not familiar with the project so it may be hard for me. I tried railway.app and it can deploy without any modifications. I thought Zeabur can do it too.

**Comment by yuaanlin** [comment, marked as answer]
You can now deploy services using Dockerfile without modify the code. Zeabur will automatically expose the port which you write in the Dockerfile.
